packages/translate/lib/testLine.tsx

```diff
@@ -89,11 +89,15 @@ const translateAssertTestLine = (
 	elements?: Elements,
 	lineResult?: TestLineResult,
 ): TestLineNode => {
+	let status = lineResult?.result;
+	if (testLine.then !== 'success' && status) {
+		status = mapStatus(lineResult?.result) === 'success' ? testLine.then : 'success';
+	}
 	const condition = translateCondition(testLine.condition, appConfig, elements, lineResult);
 
 	return <test-line
-		title={<fragment>Assert: {condition.title}{testLine.timeout ? <fragment> timeout {formatTimeout(testLine.timeout, appConfig.configVariables)}</fragment> : undefined} {testLine.then !== 'success' ? <fragment> then {translateAssertThen(testLine.then)}</fragment> : undefined}</fragment>}
-		status={lineResult?.result}
+		title={<fragment>Assert: {condition.title}{testLine.timeout ? <fragment> timeout {formatTimeout(testLine.timeout, appConfig.configVariables)}</fragment> : undefined} {testLine.then !== 'success' ? <fragment>then {translateAssertThen(testLine.then)}</fragment> : undefined}</fragment>}
```

Comment:
I think the space is supposed to be removed in different position, not before "then". Because if there is no "then" we'll get a trailing space in line.